.github/workflows/pull_request.yml

```
@@ -723,17 +723,18 @@ jobs:
       # xmlstarlet, and the BATS step needs bats. No dpkg-lock contention — this
       # is the only apt user in the job (setup-java unpacks a tarball and
       # setup-ripgrep installs a pre-built binary). Mechanism per #3779.
+      #
+      # Each network call inside the helper is bounded by a per-command timeout
+      # and retried with backoff so a stalled apt mirror or GitHub clone fails
+      # fast instead of hanging silently — a `background` failure surfaces at the
+      # `- wait:` below, and without this an indefinite hang ate the whole
+      # 50-minute job budget. `timeout-minutes` is the hard backstop.
       - name: "Install fast validation dependencies"
         shell: bash
         background: true
```

actionlint (1.7.12)

[error] unexpected key "background" for step to run shell command. expected one of "continue-on-error", "env", "id", "if", "name", "run", "shell", "timeout-minutes", "working-directory"

(syntax-check)